hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -299,11 +302,17 @@ private void refreshTimeline() throws IOException {
       schemaProvider = dataAndCheckpoint.getSchemaProvider();
     }
 
-    if ((!avroRDDOptional.isPresent()) || (avroRDDOptional.get().isEmpty())) {
-      log.info("No new data, nothing to commit.. ");
+    if (Objects.equals(checkpointStr, resumeCheckpointStr.orElse(null))) {
+      log.info("No new data, source checkpoint has not changed. Nothing to commit."
+          + "Old checkpoint=(" + resumeCheckpointStr + "). New Checkpoint=(" + checkpointStr + ")");
```

Comment:
Would it better to use log.info("No new data, source checkpoint has not changed. Nothing to commit. Old checkpoint=({}). New Checkpoint=({})", resumeCheckpointStr, checkpointStr)?

Reply:
@leesf log.info (at-least the version we use)  itself does not support variable args. It has to be log.info(String.format(xxxx)). Looks verbose IMO :) We are using the current  style everywhere in the code. So, inclined to leave it like that.

Reply:
I think slf4j has that? we are using log4j atm . HUDI-233 will track this